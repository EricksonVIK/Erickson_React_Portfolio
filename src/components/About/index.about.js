import React from "react";

function About() {
  // jsx
  return (
    <section className="about-section">
      <img
        className="profilePhoto"
        src={require("../../assets/images/Black and white profile.png")}
        alt="Chris Erickson Profile Picture"
      />
      <div className="flex-column">
        <h2>About Me</h2>
        <p className="about">
          Full Stack Developer with an in-depth knowledge of both front and
          back-end development. Currently working as an Application Consultant
          and pursuing an intensive Coding certificate from the University of
          North Carolina. Professionally, I have bolstered a diverse skill set
          including time management, communication/conflict resolution,
          strategic program development, and critical thinking/troubleshooting.
        </p>
        <p className="about">
          Excited to use a developed technical and professional skill set to
          collaborate, continue learning, and contribute as a valuable team asset.
        </p>
      </div>
    </section>
  );
}

export default About;
