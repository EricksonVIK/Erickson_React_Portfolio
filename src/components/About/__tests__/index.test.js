import React from "react";
// render -- renders components
// cleanup -- removes components
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
// import compnent to be tested
import About from "../index.about";

// ensures no leftover memory to clash
afterEach(cleanup);

describe("About component", () => {
  // First Test
  it("renders", () => {
    render(<About />);
  });
  // Second Test
  it("matches snapshot DOM node structure", () => {
      const { asFragment } = render(<About />);
      expect(asFragment()).toMatchSnapshot();
  });
});
