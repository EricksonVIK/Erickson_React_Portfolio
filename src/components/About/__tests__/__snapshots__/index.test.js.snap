// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="about-section"
  >
    <img
      alt="Chris Erickson Profile Picture"
      class="profilePhoto"
      src="Black and white profile.png"
    />
    <div
      class="flex-column"
    >
      <h2>
        About Me
      </h2>
      <p
        class="about"
      >
        Need to add text
      </p>
      <p
        class="about"
      >
        Placeholder for additional text if needed
      </p>
    </div>
  </section>
</DocumentFragment>
`;
